### PR TITLE
[5.1] Dialog: aria-label for Close button is not translated

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #42301 .

### Summary of Changes
Changes hadr coded english to a translatable string wioth fallback

### Testing Instructions
`npm run build:js` or use a prebuilt package

Enable Debug Language.
Edit an article.
Under Publishing tab > Created by, click user icon to display the modal dialog.
Using Web Developer Tools, select the Close button to inspect it.

### Actual result BEFORE applying this Pull Request
`aria-label="**Close**"
`

### Expected result AFTER applying this Pull Request
`<button type="button" aria-label="Close" class="button-close btn-close" data-button-close=""></button>
`

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 28ab937bfe5631a1f4fa1370f2debdf291bb1ef8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43058">issues.joomla.org/tracker/joomla-cms/43058</a>.</sub>

File: build/media_source/system/js/joomla-dialog.w-c.es6.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ class JoomlaDialog extends HTMLElement {
     this.popupType = this.getAttribute('type') || 'inline';
     this.textHeader = this.getAttribute('text-header') || '';
     this.iconHeader = '';
-    this.textClose = 'Close';
+    this.textClose = Joomla.Text._('JCLOSE', 'Close');;
```
Comment (Quy) [inline build/media_source/system/js/joomla-dialog.w-c.es6.js]:
```suggestion
    this.textClose = Joomla.Text._('JCLOSE', 'Close');
```

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 6bef5cd0cb74fc6ebc4a35ad2c341b79596b540d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43058">issues.joomla.org/tracker/joomla-cms/43058</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43058">issues.joomla.org/tracker/joomla-cms/43058</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**